Mathlib/Combinatorics/SimpleGraph/Hamiltonian.lean

```diff
@@ -179,4 +179,63 @@ lemma IsHamiltonian.connected (hG : G.IsHamiltonian) : G.Connected where
     exact ((p.takeUntil a a_mem).reverse.append <| p.takeUntil b b_mem).reachable
   nonempty := not_isEmpty_iff.1 fun _ ↦ by simpa using hG <| by simp [@Fintype.card_eq_zero]
 
+variable {V : Type*} [Fintype V] [DecidableEq V]
+
+/--
+A finite simple graph with at least three vertices and a bridge is not Hamiltonian.
+
+More precisely, let `G : SimpleGraph V` be a simple graph on a finite vertex type `V`.
+If `Fintype.card V ≥ 3` and there exists an edge of `G` which is a bridge,
+then `G` does not admit a Hamiltonian cycle, i.e. `¬ G.IsHamiltonian`.
+-/
+theorem IsNotHamiltonian_if_IsBridge
+    (G : SimpleGraph V)
+    (h_order : Fintype.card V ≥ 3)
```

Comment:
For technical reasons, the Mathlib convention is to always use `le` instead of `ge`. So please flip this inequality.